File: 05-select-from-a-dropdown/java/Dropdown.java
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;

public class Dropdown {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @Test
    public void dropdownTest() {
        driver.get("http://the-internet.herokuapp.com/dropdown");
        Select select = new Select(driver.findElement(By.id("dropdown")));
//        select.selectByValue("1");
        select.selectByVisibleText("Option 1");
        assertThat(select.getFirstSelectedOption().getText(), is(equalTo("Option 1")));
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }
}

File: 05-select-from-a-dropdown/csharp/Dropdown/Dropdown.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using System;
using System.Collections.Generic;

namespace Dropdown
{
    public class Dropdown
    {
        IWebDriver Driver;

        [SetUp]
        public void SetUp()
        {
            Driver = new FirefoxDriver();
        }

        [TearDown]
        public void TearDown()
        {
            Driver.Quit();
        }

        [Test]
        public void SelectFromDropdownTheHardWay()
        {
            Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/dropdown");
            IWebElement Dropdown = Driver.FindElement(By.Id("dropdown"));
            IReadOnlyCollection<IWebElement> DropdownOptions = Dropdown.FindElements(By.TagName("option"));
            foreach(IWebElement Option in DropdownOptions)
            {
                if(Option.Text.Equals("Option 1"))
                    Option.Click();
            }
            string SelectedOption = "";
            foreach (IWebElement Option in DropdownOptions)
            {
                if (Option.Selected)
                    SelectedOption = Option.Text;
            }
            Assert.That(SelectedOption.Equals("Option 1"));
        }

        [Test]
        public void SelectFromDropdownTheEasyWay()
        {
            Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/dropdown");
            SelectElement Dropdown = new SelectElement(Driver.FindElement(By.Id("dropdown")));
            Dropdown.SelectByText("Option 1");
            Assert.That(Dropdown.SelectedOption.Text.Equals("Option 1"));
        }

    }
}